### PR TITLE
Bignum mod

### PR DESCRIPTION
**Pull request by luke-bhan**
Positive modulo operator for bignum. Rather simple. Built off the mul branch. only file that is affected is modulo.rs

### CODE REVIEW

File: libslide/src/bignum/modulo.rs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    fn rem(self, rhs: Bignum) -> Bignum {
+        // we only support positive mod for now
+        if !rhs.dec.is_empty() {
+            panic!("The module fuction only supports whole numbers!");
+        }
+        let mut result = self;
+        let zero = Bignum::new("0".to_string()).unwrap();
+        while result >= zero {
+            result = result - rhs.clone();
+        }
```
Comment (ayazhafiz) [inline libslide/src/bignum/modulo.rs:17]: Can you make the implementation more efficient using binary skipping? First you subtract x, then 2x, then 4x, etc. If you go over you backup and start again from x. Lmk if this makes sense.

Also, only compare with `>`

Comment (luke-bhan) [reply libslide/src/bignum/modulo.rs:17]: yeah I definitely can. never heard of that, but thanks for introducing it to me

Comment (luke-bhan) [reply libslide/src/bignum/modulo.rs:17]: wait do you think it will be faster, because we need to them implement an extra multiply

Comment (ayazhafiz) [reply libslide/src/bignum/modulo.rs]: implement both and benchmark :-)

**Comment by luke-bhan** [comment]
how are you organizing benchamrk files?I see you have a separate folder but when I run cargo bench it does not run.

**Comment by ayazhafiz** [comment]
What;s the error?

**Comment by luke-bhan** [comment]
no error, just the benchmark results aren't produced. It only runs all the tests which are ignored.

**Comment by ayazhafiz** [comment]
Make sure you run

cargo bench —features benchmark-internals

________________________________
От: Luke Bhan <notifications@github.com>
Отправлено: Wednesday, May 27, 2020 10:42:53 PM
Кому: ayazhafiz/slide <slide@noreply.github.com>
Копия: hafiz <ayaz.hafiz.1@gmail.com>; Review requested <review_requested@noreply.github.com>
Тема: Re: [ayazhafiz/slide] Bignum mod (#74)


no error, just the benchmark results aren't produced. It only runs all the tests which are ignored.

**Comment by luke-bhan** [comment]
:) thank you

**Comment by luke-bhan** [comment]
binary skipping is faster :)

**Pull request merged**